docs/state.rst

```diff
@@ -244,3 +246,119 @@ For example:
         otherAppOtherAccountRole,
         If(otherAppOtherAccountRole.hasValue(), otherAppOtherAccountRole.value(), Bytes("none"))
     ])
+
+Box Storage
+-----------
+
+Box storage consists of key-value pairs that are stored in application's local context.
+
+The app account's minimum balance requirement (MBR) is increased by each box, and by the byte size of the box.
+If one deletes an application with outstanding boxes, the MBR is not recoverable from the deleted app account.
+
+Box sizes and names cannot be changed after initial allocation, but they can be deleted and re-allocated.
+Boxes are only visible to the app itself, other apps cannot view the app's own boxes on-chain.
```

Comment:
```suggestion
Boxes are only visible to the app itself; in other words, an app cannot read from or write to another app's boxes on-chain.
```

Reply:
it just reminds me of my advisor and I are working through paper... he's picking verbal error from each line of my writing... quite a nostalgic feeling, but probably quite painful for my advisor